src/components/PlayField/PlayField.tsx

```diff
@@ -0,0 +1,92 @@
+import styles from "./PlayField.module.css";
+import Card from "../Card/Card";
+import React from "react";
+import { initialImages } from "../../constants/initialImages";
+import { IOpenCardsItem, IPlayingFieldItem } from "../../types/types";
+import { shuffleArray } from "../../utils/util";
+
+const images = shuffleArray([...initialImages, ...initialImages]);
+const initialField: IPlayingFieldItem[] = images.map((url) => {
+  return {
+    url,
+    flip: false,
+    off: false,
+  };
+});
+
+function PlayField() {
+  const [openCards, setOpenCards] = React.useState<IOpenCardsItem[]>([]);
+  const [field, setField] = React.useState<IPlayingFieldItem[]>(initialField);
+  const handleCards = React.useCallback(() => {
+    if (openCards.length === 2) {
+      if (openCards[0].url === openCards[1].url) {
+        const newField = [...field];
+        newField[openCards[0].index] = {
+          ...field[openCards[0].index],
+          off: true,
+        };
+        newField[openCards[1].index] = {
+          ...field[openCards[1].index],
+          off: true,
+        };
+        setField(newField);
+        setOpenCards([]);
+      } else {
+        const newField = [...field];
+        newField[openCards[0].index] = {
+          ...field[openCards[0].index],
+          flip: false,
+        };
+        newField[openCards[1].index] = {
+          ...field[openCards[1].index],
+          flip: false,
+        };
+        setField(newField);
+        setOpenCards([]);
+      }
+    }
+    return;
+  }, [field, openCards]);
+  React.useEffect(() => {
+    setTimeout(() => {
+      handleCards();
```

Comment:
it seems more logical to move `if (openCards.length === 2) {` here
```js
if (openCards.length === 2) {
  handleCards();
}
```

This way you can clearly see that we want to check results only when two cards open